File: src/com/company/ATMActions.java
package com.company;

import java.io.*;
import java.nio.file.FileSystems;
import java.nio.file.Path;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Scanner;

public class ATMActions {
    private static int attempt = 0;
    private static Client bankClient;
    private static BufferedWriter writer;
    private static File statementFile;

    private static boolean isNumber(String str) {
        boolean isNumber = true;

        try {
            Integer.parseInt(str);
        } catch (NumberFormatException e) {
            isNumber = false;
        }
        return isNumber;
    }

    private static void displayOptions() {
        System.out.println("\n1. Withdraw");
        System.out.println("2. Deposit");
        System.out.println("3. Account statement");
        System.out.println("4. Quit\n");
    }

    private static String getCurrentDateTime() {
        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss");
        LocalDateTime now = LocalDateTime.now();
        return dtf.format(now);
    }

    private static void ATMMenu(Scanner scanner) {
        int amount;

        System.out.print("\nPlease, choose one of the choice proposed below :");
        displayOptions();
        String operationChoice = scanner.nextLine();

        switch (operationChoice) {
            case "1":
                System.out.print("How much do you want to withdraw (in euro) ? : ");
                try {
                    amount = Integer.valueOf(scanner.nextLine());
                    if (bankClient.getBankAccount().withdraw(amount))
                        writer.write(getCurrentDateTime() + ": Withdraw of " + amount + "€\n\n"); //Peut etre faire une fonction qui ecrit dans le fichier et la mettre aussi en bas
                } catch(NumberFormatException | IOException e) {
                    System.out.println("Please, enter a valid amount");
                }
                break;
            case "2":
                System.out.println("How much do you want to deposit (in euro) ? : ");
                try {
                    amount = Integer.valueOf(scanner.nextLine());
                    if (bankClient.getBankAccount().deposit(amount))
                        writer.write(getCurrentDateTime() + ": Deposit of " + amount + "€\n\n");
                } catch(NumberFormatException | IOException e) {
                    System.out.println("Please, enter a valid amount");
                }
                break;
            case "3":
                System.out.println("Please wait, we are printing your account statement...\n");
                //bankClient.getBankAccount().getAccountStatement();
                getAccountStatement();
                System.out.println("Current account balance : " + bankClient.getBankAccount().getAvailableAmount() + "€");
                break;
            case "4":
                System.out.println("Goodbye " + bankClient.getName());
                System.out.println("Don't forget to get your credit card");
                try {
                    writer.close();
                    statementFile.delete();
                } catch (IOException e) {
                    e.printStackTrace();
                }
                return;
            default:
                System.out.println("Please, select one of the option displayed on the screen");
                break;
        }

        ATMMenu(scanner);
    }

    private static void pinCodeVerification(Scanner scanner) {
        System.out.print("Hello " + bankClient.getName() + ", please enter your pin code : ");
        String pinVerification = scanner.nextLine();
        if (pinVerification.equals(bankClient.getPinCode()))
            ATMMenu(scanner);
        else {
            System.out.println("Wrong pin code\n");
            attempt += 1;
            if (attempt == 3) {
                System.out.println("You have enter a wrong pin 3 times, " +
                        "your credit card has been blocked.\nPlease contact your bank.");
                return;
            }
            pinCodeVerification(scanner);
        }
    }

    static void pinCodeCreation(String name, Scanner scanner) {
        System.out.print("PLease choose a pin code for your credit card : ");
        String pinCode = scanner.nextLine();
        if (!isNumber(pinCode)) {
            System.out.println("Enter a valid pin code please");
            pinCodeCreation(name, scanner);
        }

        BankAccount bankAccount = new BankAccount(0);
        bankClient = new Client(name, pinCode, bankAccount);
        createStatementFile();

        pinCodeVerification(scanner);
    }

    private static void createStatementFile() {
        Path path = FileSystems.getDefault().getPath("./resources").toAbsolutePath();

        try {
            statementFile = new File(path + "/" + "BankStatement.txt");
            if (statementFile.createNewFile()) {
                writer = new BufferedWriter(new FileWriter(statementFile));
                writer.write(bankClient.getName() + "'s Account Statement\n\n");
            }
        } catch(IOException e) {
            e.printStackTrace();
        }
    }

    private static void getAccountStatement() {
        try {
            writer.flush();
            FileReader fr = new FileReader(statementFile);
            BufferedReader br = new BufferedReader(fr);
            String line;

            while ((line = br.readLine()) != null) {
                System.out.println(line);
            }
            fr.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: csharp/BankATM/ATMActions.cs
using System;

namespace BankATM
{
    public class ATMActions
    {
        private static int attempt = 0;
        private static Client bankClient;
        private static string accountStatement = "";

        private static bool isNumber(string str)
        {
            int number;

            return int.TryParse(str, out number);
        }

        private static void displayOptions()
        {
            Console.WriteLine("\n1. Withdraw");
            Console.WriteLine("2. Deposit");
            Console.WriteLine("3. Account statement");
            Console.WriteLine("4. Quit\n");
        }

        private static string getCurrentDate()
        {
            return DateTime.Now.ToString("dd/MM/yyyy");
        }

        private static void ATMMenu()
        {
            int amount;

            Console.Write("\nPlease, choose one of the choice proposed below :");
            displayOptions();
            String operationChoice = Console.ReadLine();

            switch (operationChoice)
            {
                case "1":
                    Console.Write("How much do you want to withdraw (in euro) ? : ");
                    try
                    {
                        amount = Int32.Parse(Console.ReadLine());
                        int amountLength = amount.ToString().Length;

                        if (bankClient.getBankAccount().withdraw(amount))
                            addOperationToSTatement("\n WITHDRAW   | " + getCurrentDate() + " | " + amount +
                                printSpaces(amount) + " | " + bankClient.getBankAccount().getAvailableAmount());
                    }
                    catch (FormatException e)
                    {
                        Console.WriteLine("Please, enter a valid amount");
                    }
                    break;
                case "2":
                    Console.WriteLine("How much do you want to deposit (in euro) ? : ");
                    try
                    {
                        amount = Int32.Parse(Console.ReadLine());

                        if (bankClient.getBankAccount().deposit(amount))
                            addOperationToSTatement("\n DEPOSIT    | " + getCurrentDate() + " | " + amount +
                                printSpaces(amount) + " | " + bankClient.getBankAccount().getAvailableAmount());
                    }
                    catch (FormatException e)
                    {
                        Console.WriteLine("Please, enter a valid amount");
                    }
                    break;
                case "3":
                    Console.WriteLine("Please wait, we are printing your account statement...\n");
                    Console.WriteLine(getAccountStatement());
                    break;
                case "4":
                    Console.WriteLine("Goodbye " + bankClient.getName());
                    Console.WriteLine("Don't forget to get your credit card");
                    return;
                default:
                    Console.WriteLine("Please, select one of the option displayed on the screen");
                    break;
            }

            ATMMenu();
        }

        private static void pinCodeVerification()
        {
            Console.Write("Hello " + bankClient.getName() + ", please enter your pin code : ");
            string pinVerification = Console.ReadLine();
            if (String.Equals(pinVerification, bankClient.getPinCode()))
                ATMMenu();
            else
            {
                Console.WriteLine("Wrong pin code\n");
                attempt += 1;
                if (attempt == 3)
                {
                    Console.WriteLine("You have enter a wrong pin 3 times, " +
                            "your credit card has been blocked.\nPlease contact your bank.");
                    return;
                }
                pinCodeVerification();
            }
        }

        public static void pinCodeCreation(string name)
        {
            Console.Write("PLease choose a pin code for your credit card : ");
            string pinCode = Console.ReadLine();
            if (!isNumber(pinCode))
            {
                Console.WriteLine("Enter a valid pin code please");
                pinCodeCreation(name);
            }

            BankAccount bankAccount = new BankAccount(0);
            bankClient = new Client(name, pinCode, bankAccount);

            accountStatement += " Operation  | Date       | Amount     | Balance    \n" +
                " -----------------------------------------------";

            pinCodeVerification();
        }

        private static void addOperationToSTatement(string operation)
        {
            accountStatement += operation;
        }

        private static string getAccountStatement()
        {
            return accountStatement;
        }

        private static string printSpaces(int amount)
        {
            int amountLength = amount.ToString().Length;
            string spaces = "";

            for (int i = 0; i <= 9 - amountLength; i++)
            {
                spaces += " ";
            }

            return spaces;
        }
    }
}
